pageserver/src/page_service.rs

```diff
@@ -447,6 +539,93 @@ impl PageServerHandler {
         Ok(())
     }
 
+    fn handle_import_basebackup(
+        &self,
+        pgb: &mut PostgresBackend,
+        tenant_id: ZTenantId,
+        timeline_id: ZTimelineId,
+        base_lsn: Lsn,
+        _end_lsn: Lsn,
+    ) -> anyhow::Result<()> {
+        thread_mgr::associate_with(Some(tenant_id), Some(timeline_id));
+        let _enter =
+            info_span!("import basebackup", timeline = %timeline_id, tenant = %tenant_id).entered();
+
+        // Create empty timeline
+        info!("creating new timeline");
+        let repo = tenant_mgr::get_repository_for_tenant(tenant_id)?;
+        let timeline = repo.create_empty_timeline(timeline_id, Lsn(0))?;
+        let repartition_distance = repo.get_checkpoint_distance();
+        let mut datadir_timeline =
+            DatadirTimeline::<LayeredRepository>::new(timeline, repartition_distance);
+
+        // TODO mark timeline as not ready until it reaches end_lsn?
+        // This would prevent compute node from connecting to it and writing conflicting wal.
```

Comment:
I suspect that compute node won't be able to connect anyway.
To connect, it needs to receive basebackup at some lsn and get the data, which is invisible until `    modification.commit()?;`

we can test it using sleep failpoint here.

Reply:
Possible scenario:
1. We import the base
2. We start importing wal
3. But before that happens, compute connects and writes wal

I'll clarify. No need to fix now, since it doesn't affect pageserver->pageserver migration

Reply:
clarified with comment